plugins/USBPrinting/USBPrinterOutputDevice.py

```diff
@@ -566,7 +566,7 @@ def _sendNextGcodeLine(self):
             if line == "M0" or line == "M1":
                 line = "M105"  # Don't send the M0 or M1 to the machine, as M0 and M1 are handled as an LCD menu pause.
             if ("G0" in line or "G1" in line) and "Z" in line:
-                z = float(re.search("Z([0-9\.]*)", line).group(1))
+                z = float(re.search("Z([-0-9\.]*)", line).group(1))
```

Comment:
For Cura, the validation of the gcode is done by the CuraEngine already, right?
I'm a fan of _as simple as possible regex_ for extracting values from a known upstream and relying on the float to throw ValueErrors for any nonsensical matches.

Maybe we could make the logging for the exception more descriptive so that it is specifically known this is a G1 Z<value> conversion error?

The M0, etc, stuff seems out of place in that block. I can do a separate PR refactoring that exception block and logging if that sounds like something that is wanted...?